### ISSUE TITLE
Fix the TestServer overloads

### ISSUE THREAD

**Issue by davidfowl** [aspnet/Hosting#518]
They are a mess https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNet.TestHost/TestServer.cs#L36-L96. We need to change this API to look more like WebHostBuilder so it can be configured without adding overloads.

**Comment by Tratcher** [comment]
Maybe this can consolidate down to something like:
```
var server = TestServer.Create(webHostBuilder =>
{
  webHostBuilder.UseStartup<MyStartup>();
});
```

**Comment by JunTaoLuo** [comment]
Will be addressed by #525
